pkg/kubelet/config/config.go

```diff
@@ -416,13 +422,26 @@ func podsDifferSemantically(existing, ref *api.Pod) bool {
 	return true
 }
 
-// checkAndUpdatePod updates existing if ref makes a meaningful change and returns true, or
-// returns false if there was no update.
-func checkAndUpdatePod(existing, ref *api.Pod) bool {
+// checkAndUpdatePod updates existing, and:
+//   * if ref makes a meaningful change, returns needUpdate=true
+//   * if ref makes no meaningful change, but changes the pod status, returns needReconcile=true
+//   * else return both false
+//   Now, needUpdate and needReconcile should never be both true
+//   NOTE: (random-liu) Any suggestions for the code format here, what is the best practice?
```

Comment:
(optional) I think you're using 2 bools as a tri-state. You could add a new "enum" for this, but that might be overkill. Alternatively, you could just reuse `kubetypes.PodOperation` as the "update type" (would need to be a pointer for the nil case). I'm not sure either is better, just listing some possibilities...